packages/suite-base/src/components/PanelExtensionAdapter/renderState.test.ts

```diff
@@ -1125,4 +1045,131 @@ describe("renderState", () => {
     expect(checkRenderedConfig).toHaveBeenCalled();
     expect(checkRenderedConfig.mock.calls.at(-1)).toEqual([{ test: false }]);
   });
+
+  it("should update renderStateField when watchedFields contains parameters", async () => {
+    const buildRenderState = initRenderStateBuilder();
+    const stableConversionInputs = {
+      sortedTopics: [],
+      subscriptions: [{ topic: "test" }],
+      messageConverters: [],
+    };
+
+    const state = buildRenderState({
+      watchedFields: new Set(["parameters"]),
+      playerState: undefined,
+      appSettings: undefined,
+      currentFrame: undefined,
+      colorScheme: undefined,
+      globalVariables: {},
+      hoverValue: undefined,
+      sharedPanelState: {},
+      ...stableConversionInputs,
+    });
+
+    expect(state).toEqual({
+      parameters: new Map(),
+    });
+  });
+
+  it("should renderStateField when activeData contains parameters", async () => {
+    const buildRenderState = initRenderStateBuilder();
+    const stableConversionInputs = {
+      sortedTopics: [],
+      subscriptions: [{ topic: "test" }],
+      messageConverters: [],
+    };
+    const activeData = PlayerStateBuilder.activeData({
+      parameters: BasicBuilder.genericMap<string>(BasicBuilder.string),
+    });
+    const playerState = PlayerStateBuilder.playerState({ activeData });
+    const state = buildRenderState({
+      watchedFields: new Set(["parameters"]),
+      playerState,
+      appSettings: undefined,
+      currentFrame: undefined,
+      colorScheme: undefined,
+      globalVariables: {},
+      hoverValue: undefined,
+      sharedPanelState: {},
+      ...stableConversionInputs,
+    });
+
+    expect(state).toEqual({
+      parameters: activeData.parameters,
+    });
+  });
```

Comment:
Second example using `setup()`
```suggestion
  it("should renderStateField when activeData contains parameters", async () => {
    const { buildRenderState, input } = setup();
    const activeData = PlayerStateBuilder.activeData({
      parameters: BasicBuilder.genericMap<string>(BasicBuilder.string),
    });
    const playerState = PlayerStateBuilder.playerState({ activeData });
    _.merge(input, {
      messageConverters: [],
      playerState,
      sortedTopics: [],
      subscriptions: [BasicBuilder.genericMap],
      watchedFields: new Set(["parameters"]),
    });
    const state = buildRenderState(input);

    expect(state).toEqual({
      parameters: activeData.parameters,
    });
  });
  ```